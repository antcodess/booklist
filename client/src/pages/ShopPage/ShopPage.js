import React from 'react';
import BreadCrumbNav from '../../components/BreadCrumbNav/BreadCrumbNav';
import CategoriesOverview from '../../components/CategoriesOverview/CategoriesOverview';
import './ShopPage.scss';

export const ShopPage = ({ match }) => {
  
  const shopPath = match.path.replace('/', '');
  const shopTitle = shopPath[0].toUpperCase() + shopPath.slice(1);

  return (
    <React.Fragment>
      <BreadCrumbNav title={shopTitle} />
      <section className='shop-page'>
        <h2>{shopTitle}</h2>
        <CategoriesOverview />
      </section>
    </React.Fragment>
  );
};

export default ShopPage;