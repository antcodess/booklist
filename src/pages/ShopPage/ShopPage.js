import React from 'react';
import BreadCrumbNav from '../../components/BreadCrumbNav/BreadCrumbNav';
import CategoryOverview from '../../components/CategoryOverview/CategoryOverview';

const ShopPage = ({ match }) => {
  const shopPath = match.path.replace('/', '');
  const shopTitle = shopPath[0].toUpperCase() + shopPath.slice(1);
  return (
    <React.Fragment>
      <BreadCrumbNav title={shopTitle} />
      <CategoryOverview />
    </React.Fragment>
  );
};

export default ShopPage;