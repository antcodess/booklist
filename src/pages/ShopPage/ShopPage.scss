.shop-page {
  h2 {
    text-align: left;
    margin-top: 20px;
    font-size: 2rem; 
  }
}