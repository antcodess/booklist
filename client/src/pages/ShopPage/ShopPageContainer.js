import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';
import { selectIsProductFetching } from '../../redux/shop/shop.selectors';

import WithSpinner from '../../components/WithSpinner/WithSpinner';
import ShopPage from './ShopPage';

const mapStateToProps = createStructuredSelector({
  isLoading: selectIsProductFetching,
});

const ShopPageContainer = compose(
  connect(mapStateToProps),
  WithSpinner
)(ShopPage);

export default ShopPageContainer;