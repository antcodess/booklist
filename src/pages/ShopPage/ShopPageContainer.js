import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';
import { selectIsProductFetching } from '../../redux/shop/shop.selectors';

import Spinner from '../../components/Spinner/Spinner';
import ShopPage from './ShopPage';

const mapStateToProps = createStructuredSelector({
  isLoading: selectIsProductFetching,
});

const ShopPageContainer = compose(
  connect(mapStateToProps),
  Spinner
)(ShopPage);

export default ShopPageContainer;