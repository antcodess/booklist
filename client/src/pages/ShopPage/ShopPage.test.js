import React from 'react';
import { shallow } from 'enzyme';
import { ShopPage } from './ShopPage';

describe('ShopPage Component', () => {
  let wrapper;
  let mockMatch;

  beforeEach(() => {
    mockMatch = {
      path: '/shop'
    }

    const mockProps = {
      match: mockMatch
    };
    wrapper = shallow(<ShopPage { ...mockProps } />);
  });

  it('should render ShopPage Component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});