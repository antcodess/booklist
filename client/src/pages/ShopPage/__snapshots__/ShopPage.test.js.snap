// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopPage Component should render ShopPage Component 1`] = `ShallowWrapper {}`;
